frontend/apps/reader/modules/readerpaging.lua

```diff
@@ -241,8 +242,7 @@ function ReaderPaging:addToMainMenu(menu_items)
         text = _("Invert page turn taps and swipes"),
         checked_func = function() return self.inverse_reading_order end,
         callback = function()
-            self.inverse_reading_order = not self.inverse_reading_order
-            self:setupTapTouchZones()
+            self:onToggleReadingOrder()
```

Comment:
In reply to https://github.com/koreader/koreader/pull/5423#issuecomment-535194152

> that would not be enough because the code would still have to be be duplicated in the menu code.

This line here is the tiniest bit faster than `self.ui:handleEvent(Event:new("ToggleReadingOrder"))` (or something along those lines) because in the current scenario there's no need to bubble the event through various widgets.

Reply:
i was trying to deduplicate the code, if the speed and not going back and forth between modules is better i will change it back, i am new to the code and not sure of conventions, if there are any.

Reply:
And I'm asking for it but not seeing any yet. :-P

```lua
function Reader**Gesture**:onToggleReadingOrder()
-- this is where the line I suggested earlier goes
    local document_module = self.ui.document.info.has_pages and self.ui.paging or self.ui.rolling
-- and most of the selfs should be replaced by the reference to the module
    self.inverse_reading_order = not self.inverse_reading_order
    self:setupTouchZones()
    -- Needed to reset the touch zone overrides
-- this one however can then become self:onReSetupGesture
    self.ui:handleEvent(Event:new("ReSetupGesture"))
    UIManager:show(Notification:new{
        text = self.inverse_reading_order and _("RTL page turning.") or _("LTR page turning."),
        timeout = 2.5,
    })
    return true
end

Reply:
done